spring-integration-core/src/main/java/org/springframework/integration/config/xml/PointToPointChannelParser.java

```diff
@@ -55,13 +56,19 @@ protected BeanDefinitionBuilder buildBeanDefinition(Element element, ParserConte
 			builder = BeanDefinitionBuilder.genericBeanDefinition(QueueChannel.class);
 			boolean hasStoreRef = this.parseStoreRef(builder, queueElement, channel, false);
 			boolean hasQueueRef = this.parseQueueRef(builder, queueElement);
-			if (!hasStoreRef) {
+			if (!hasStoreRef || !hasQueueRef) {
 				boolean hasCapacity = this.parseQueueCapacity(builder, queueElement);
 				if (hasCapacity && hasQueueRef) {
 					parserContext.getReaderContext().error(
 							"The 'capacity' attribute is not allowed" + " when providing a 'ref' to a custom queue.",
 							element);
 				}
+				if (hasCapacity && hasStoreRef) {
+					parserContext.getReaderContext().error(
+							"The 'capacity' attribute is not allowed"
+									+ " when providing a 'message-store' to a custom MessageGroupStore.",
```

Comment:
It's late! I'm on it already during merge with some other simple code style polishing around.
BTW, you have not to do any code style refactoring during your PR. That will simplify our review life :smile: